### DISCUSSION TITLE
Enhancements to PR, Issue, and Contribution workflows

### DISCUSSION THREAD

**Discussion by joshuaboniface** [jellyfin/jellyfin-meta#40] [General]
A couple ideas to come out of chat in the last few days that are worth recording for further input.

1. Let's improve documentation for new contributors as to our contribution/GitHub workflows. I'm thinking topics such as "open discussions [in the relevant repo, or jellyfin-meta if covering multiple components] before contributing code to get feedback on ideas", "how to handle Fider", "how to get support from the team in Chat", and "how to handle reviews, delays, and merging". In addition, of course, we should really get code style pinned down especially for the core server: I know @cvium, @Bond-009, and @crobibero all have some particular ideas in this regard so getting those actually written would go a long way to streamlining PR reviews by, hopefully, getting contributors to "do things right" from the beginning.

2. Let's extend stalebot to auto-close PRs that have had a merge conflict for greater than some set period to avoid these obsolete and clearly-not-important PRs from clogging up the list. I think 3 or 6 months is reasonable there. In addition I think the action should ping the author explicitly when adding the label since I don't believe simply adding a label triggers an email/notification alert.

3. Further, I think we could go one step further in the opposite direction, and see if we can have an action auto-ping teams for review if a PR sits open and idle for too long, say a month and every month after until merged or closed. This will hopefully get someone's attention and avoid PRs being missed and left languishing for many months.

**Comment by thornbill** [comment]
We do something similar to point 2 for the expo app, but with the "question" label, so we could probably get some ideas from there.

**Comment by oddstr13** [comment]
I can confirm that simply adding a label does not trigger email notifications for me, and I've had issues closed and locked on me in different projects due to this.

**Comment by thornbill** [comment]
I couldn't really find a good out of the box solution for point 3. We might need to use something like https://github.com/actions/github-script to write it ourselves.

**Reply by thornbill** [reply]
Looks like https://github.com/marketplace/actions/review-reminder would do exactly what we want here.
